vendor/github.com/rubiojr/go-vhd/vhd/vhd.go

```diff
@@ -324,8 +327,14 @@ func (vhd *VHD) PrintExtraHeader() {
 	fmtField("Parent timestamp", fmt.Sprintf("%s", t))
 
 	fmtField("Reserved", hexs(header.Reserved[:]))
-	parentName := utf16BytesToString(header.ParentUnicodeName[:],
-		binary.BigEndian)
+	parentNameBytes, _, err := transform.Bytes(
+		unicode.UTF16(unicode.BigEndian, unicode.IgnoreBOM).NewDecoder(),
+		header.ParentUnicodeName[:],
+	)
+	if err != nil {
+		panic(err)
```

Comment:
where does this data come from? does this open up a DOS vector where invalid data can cause the controller manager to panic?

Reply:
> does this open up a DOS vector where invalid data can cause the controller manager to panic?

I'm not familiar with Kubernetes so I can't answer that but:

* The data comes from a VHD header, so a malformed VHD header could trigger that. This helper doesn't really validate the VHD header format in any way so it'd be trivial to cause that even without the changes reviewed here.
* It's a helper function used by the CLI (`go-vhd info`) to print VHD header values to stdout.

Hope that helps.

Reply:
that's helpful, thanks. I double checked and kube doesn't call PrintInfo() or PrintExtraHeader() anywhere